### PR TITLE
feat(medusa,pricing,types): added get endpoints for pricing rule types

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds list endpoint for rule types
- adds get endpoint for rule types

### CODE REVIEW

File: packages/pricing/src/migrations/Migration20240312163523.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Migration } from "@mikro-orm/migrations"
```
Comment (olivermrbl) [inline packages/pricing/src/migrations/Migration20240312163523.ts]: todo: Include in "master" migration

File: packages/pricing/src/migrations/Migration20240312163523.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export class Migration20240312163523 extends Migration {
+  async up(): Promise<void> {
```
Comment (olivermrbl) [inline packages/pricing/src/migrations/Migration20240312163523.ts]: question: Is the new index missing here?

**Review by olivermrbl — APPROVED** [review]
Nice work, minor comments 💪

File: packages/pricing/src/migrations/Migration20240312163523.ts (hunk 2 above)
Comment (riqwan) [reply packages/pricing/src/migrations/Migration20240312163523.ts]: no, its already present in the first migration. It just got missed in the model.

File: packages/pricing/src/migrations/Migration20240312163523.ts (hunk 1 above)
Comment (riqwan) [reply packages/pricing/src/migrations/Migration20240312163523.ts]: done

**Pull request merged**